quests.md

```diff
@@ -0,0 +1,272 @@
+```json
+{
+  "throw_id": int
+}
+```
+This can have one of the following
+
+| ID | Name      |
+|----|-----------|
+| 0  | Normal    |
+| 1  | Nice      |
+| 2  | Great     |
+| 3  | Excellent |
+
+For `WITH_THROW_TYPE_IN_A_ROW`, increase the `amount` field on the condition
+
+#### Item (type `11`)
+```json
+{
+  "item_id": int
+}
+```
+See [item array](#item-array) for id options
+
+
+
+
+
+## Rewards
+
+This spec will cover most of the requirements, but when more information is needed, the `detail` field should be 
+included
+
+Otherwise, the `detail` field will not be checked
+
+| Key           | Type   | Description                                                                           |
+|---------------|--------|---------------------------------------------------------------------------------------|
+| type          | int    | See [the reward types array](#rewards-type-array) for options                         |
+| id            | int    | A generic id representing an object, based on `type`                                  |
+| amount        | int    | A generic amount representing how many must be acquired or completed, based on `type` |
+| detail        | object | See [the rewards detail section](#rewards-detail) for options                         |
+
+
+```json
+{
+  "type": int,
+  "id": int,
+  "amount": int,
+  "detail": {...}
+}
+```
+
+### Rewards Type Array
+This came directly out of the protos, and the `type` int field for every reward should represent one of these
+
+Keep in mind that all values will be accepted, but not all will be valid, even values in this list
+```
+enum Type {
+    UNSET = 0;
+    EXPERIENCE = 1;
+    ITEM = 2;
+    STARDUST = 3;
+    CANDY = 4;
+    AVATAR_CLOTHING = 5;
+    QUEST = 6;
+    POKEMON_ENCOUNTER = 7;
+}
+```
+
+
+### Rewards detail
+Rewards details can have a multitude of keys based on the type of reward (see [Rewards Type Array](#rewards-type-array))
+
+All types listed in the enum are valid, but only a few are useful, and even fewer need more details than what's already
+been defined
+
+#### Monster (type `7`):
+This section should have most required fields as a normal monster webhook, excluding encounter information.
+```json
+{
+    "pokemon_id": int,
+    "form": int,
+    "gender": int,
+    "costume_id": int
+}
+```
+
+* In the case of the reward monster being a ditto, just send the pokemon id of ditto
+* Along with these, the `shiny` field may also be added here for possible later use, as a `boolean` type
+    * This is because nobody really knows what it's for. It is assumed this was used for Squirtle community day, when a 
+        quest from a specific stop would guarantee a shiny for everybody that got the quest from that stop.
+
+#### Candy (type `4`)
+The only extra information that's needed is the ID of the monster that you would receive candy for 
+```json
+{
+  "pokemon_id": int
+}
+```
+* ID of `0` will assume the Rare Candy item
+
+
+
+
+## Item array
```

Comment:
Maybe we should consider moving the ID definitions into a separate folder? 

For example monster id's will likely be used, I'm not sure it make sense to continue to list them in multiple places.

Reply:
I listed this mostly because PA doesn't currently have a game item definition so far

Reply:
Right, I'm not saying it's not needed - I'm saying that rather rather than say "this depends on PA's definitions" we should aim for "here is the definitions, PA also implements them".